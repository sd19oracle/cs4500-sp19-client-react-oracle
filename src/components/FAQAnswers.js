import {Link} from "react-router-dom";
import React from "react";

const FAQAnswers = ({faqAnswers}) => <div>
  <h3>FAQ Answers</h3>
  <table className="table">
    <thead>
    <tr>
      <td>ID</td>
      <td>Question ID</td>
      <td>Question</td>
      <td>Answer</td>
    </tr>
    </thead>
    <tbody>
    {
      faqAnswers
        .map(faqAnswer =>
          <tr key={faqAnswer.id}>
            <td>
              <Link to={`/admin/faq-answers/${faqAnswer.id}`}>{faqAnswer.id}</Link>
            </td>
            <td>{faqAnswer.questionId}</td>
            <td>{faqAnswer.question}</td>
            <td>{faqAnswer.answer}</td>
          </tr>
        )
    }
    </tbody>
  </table>
</div>;

export default FAQAnswers;
