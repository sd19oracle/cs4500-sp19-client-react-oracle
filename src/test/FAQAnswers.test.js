import React from "react";
import renderer from "react-test-renderer";
import faas from "../data/frequently_asked_answers.mock";
import FAQAnswers from "../components/FAQAnswers";
import {MemoryRouter, Route, StaticRouter} from "react-router-dom";

test("should render the faq answers", () => {
  const component = renderer.create(
    <MemoryRouter>
      <FAQAnswers faqAnswers={faas}/>
    </MemoryRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  let root = component.root;
  expect(root.findAllByType("a").length).toBe(4);
  expect(root.findAllByType("td").length).toBe(20);
});

test("renders the answers given", () => {
  const component = renderer.create(
    <MemoryRouter>
      <FAQAnswers faqAnswers={faas.slice(1)}/>
    </MemoryRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  let root = component.root;
  expect(root.findAllByType("a").length).toBe(3);
  expect(root.findAllByType("td").length).toBe(16);
});
