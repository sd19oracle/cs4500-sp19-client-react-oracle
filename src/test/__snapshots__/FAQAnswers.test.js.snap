// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the answers given 1`] = `
<div>
  <h3>
    FAQ Answers
  </h3>
  <table
    className="table"
  >
    <thead>
      <tr>
        <td>
          ID
        </td>
        <td>
          Question ID
        </td>
        <td>
          Question
        </td>
        <td>
          Answer
        </td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <a
            href="/admin/faq-answers/2"
            onClick={[Function]}
          >
            2
          </a>
        </td>
        <td>
          2
        </td>
        <td>
          Are you getting the hang of this Chris?
        </td>
        <td>
          No
        </td>
      </tr>
      <tr>
        <td>
          <a
            href="/admin/faq-answers/3"
            onClick={[Function]}
          >
            3
          </a>
        </td>
        <td>
          2
        </td>
        <td>
          Are you getting the hang of this Chris?
        </td>
        <td>
          Maybe
        </td>
      </tr>
      <tr>
        <td>
          <a
            href="/admin/faq-answers/4"
            onClick={[Function]}
          >
            4
          </a>
        </td>
        <td>
          3
        </td>
        <td>
          test
        </td>
        <td>
          So
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`should render the faq answers 1`] = `
<div>
  <h3>
    FAQ Answers
  </h3>
  <table
    className="table"
  >
    <thead>
      <tr>
        <td>
          ID
        </td>
        <td>
          Question ID
        </td>
        <td>
          Question
        </td>
        <td>
          Answer
        </td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <a
            href="/admin/faq-answers/1"
            onClick={[Function]}
          >
            1
          </a>
        </td>
        <td>
          1
        </td>
        <td>
          Do you like pie?
        </td>
        <td>
          Yes
        </td>
      </tr>
      <tr>
        <td>
          <a
            href="/admin/faq-answers/2"
            onClick={[Function]}
          >
            2
          </a>
        </td>
        <td>
          2
        </td>
        <td>
          Are you getting the hang of this Chris?
        </td>
        <td>
          No
        </td>
      </tr>
      <tr>
        <td>
          <a
            href="/admin/faq-answers/3"
            onClick={[Function]}
          >
            3
          </a>
        </td>
        <td>
          2
        </td>
        <td>
          Are you getting the hang of this Chris?
        </td>
        <td>
          Maybe
        </td>
      </tr>
      <tr>
        <td>
          <a
            href="/admin/faq-answers/4"
            onClick={[Function]}
          >
            4
          </a>
        </td>
        <td>
          3
        </td>
        <td>
          test
        </td>
        <td>
          So
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
